dandiapi/api/views/asset.py

```diff
@@ -291,14 +300,12 @@ def create(self, request, versions__dandiset__pk, versions__version):
                 )
                 asset_blob = None
             zarr_archive = None
-        elif 'blob_id' not in serializer.validated_data and 'zarr_id' in serializer.validated_data:
+        else:
```

Comment:
I actually think the explicit distinction between `blob_id` and `zarr_id` in these if/elif statements makes the intent here clearer. Also, if we ever have additional "special" assets like zarr, I fear that having the `else` block implicitly mean it's a zarr would confuse things

What do you think of keeping 
```
elif 'blob_id' not in serializer.validated_data and 'zarr_id' in serializer.validated_data`:
	.....
```
And then changing the `else` block on line 300 to be
```
else:
	raise NotImplementedError('.....')
```

(Obviously this would apply to line 395+ too)

Reply:
That's a good point, I'll change this